integration_test/third_party_apps_test.go

```diff
@@ -363,6 +364,75 @@ func runSingleTest(ctx context.Context, logger *logging.DirectoryLogger, vm *gce
 	return nonRetryable, nil
 }
 
+func modifiedFiles(t *testing.T) []string {
+	cmd := exec.Command("git", "diff", "--name-only", "upstream/master")
+	out, err := cmd.Output()
+	if err != nil {
+		t.Fatalf("got error calling `git diff`: %v", err)
+	}
+
+	return strings.Split(string(out), "\n")
+}
+
+func determineImpactedApps(mf []string) map[string]bool {
+	impactedApps := make(map[string]bool)
+	for _, f := range mf {
+		if strings.HasPrefix(f, "apps/") {
+			// File names: apps/<appname>.go
+			f := strings.TrimPrefix(f, "apps/")
+			f = strings.TrimSuffix(f, ".go")
+			impactedApps[f] = true
```

Comment:
how does this deal with modified files like common_logging_processors.go? https://github.com/GoogleCloudPlatform/ops-agent/blob/master/apps/common_logging_processors.go

I think you need to check that the app is valid by checking it against integration_test/third_party_apps_data/applications/<app_name>. Unless that is being done later somewhere, in which case I missed it. Otherwise won't it try to run an app called common_logging_processors?

Reply:
This is a gap.  Let me figure out what to do here.